src/sqlfluff/core/linter/linted_path.py

```diff
@@ -0,0 +1,118 @@
+"""Defines the LintedPath class.
+
+This stors the idea of a collection of linted files at a single start path
+"""
+
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+    overload,
+)
+from typing_extensions import Literal
+
+
+from sqlfluff.core.errors import (
+    CheckTuple,
+)
+from sqlfluff.core.parser.segments.base import BaseSegment
+
+from sqlfluff.core.linter.linted_file import LintedFile
+
+
+class LintedPath:
+    """A class to store the idea of a collection of linted files at a single start path."""
+
+    def __init__(self, path: str) -> None:
+        self.files: List[LintedFile] = []
+        self.path: str = path
+
+    def add(self, file: LintedFile) -> None:
+        """Add a file to this path."""
+        self.files.append(file)
+
+    @overload
+    def check_tuples(self, by_path: Literal[False]) -> List[CheckTuple]:
+        """Return a List of CheckTuples when by_path is False."""
+        ...

```

Comment:
I haven't seen this before -- what does the "..." do here?

Reply:
This is for overloading. https://python.readthedocs.io/en/stable/library/typing.html#typing.overload

Reply:
Got it -- thanks!